docs/source/security/cors.mdx

```diff
@@ -0,0 +1,108 @@
+};
+
+server.applyMiddleware({
+  app,
+  cors: corsOptions,
+  path: "/graphql",
+});
+```
+
+### Configuring origins
+
+CORS uses [specific HTTP request headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#the_http_response_headers) as part of its protocol; the most well-known is [`Access-Control-Allow-Origin`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin). The `Access-Control-Allow-Origin` header enables a server to dictate which origins can use scripts to access that server's resources. Using [`cors`](https://github.com/expressjs/cors#configuration-options) middleware, you can configure the `Access-Control-Allow-Origin` header using the `origins` option.
+
+By default, the batteries-included `apollo-server` package uses the wildcard value (`*`) to allow requests from any origin _without credentials_ to access its served responses. This works great in development, allowing you to delay configuring CORS until your application is ready for production. You can also [remove CORS middleware entirely](../api/apollo-server#cors), though allowing all traffic to your server can lead to security implications.
+
+> If you need to pass credentials to your server (e.g., via cookies or an HTTP header), you can't use the wildcard value (`*`) for your origin. You _must_ provide specific origins; see [Passing credentials with CORS](#passing-credentials-with-cors) for examples.
+
+When your server is in production, we strongly recommend listing the specific origins that should have access to your server:
+
+```js
+//  With the batteries-included Apollo Server
+const server = new ApolloServer({
+  typeDefs,
+  resolvers,
+  cors: {
+    origin: ["http://localhost:3000", "https://studio.apollographql.com"]
+  },
+});
+```
+
+> Note if you plan to use Apollo Studio you should include https://studio.apollographql.com in your list of valid origins.
+
+## Passing credentials with CORS
+
+If your cross-site requests [include](https://developer.mozilla.org/en-US/docs/Web/API/Request/credentials) a user's credentials (e.g., cookies or HTTP Authentication headers), you might need to modify your CORS configuration.
+
+You can enable credentials with CORS by setting the [`Access-Control-Allow-Credentials`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Credentials) HTTP header to `true`.
+
+Using `cors` middleware, we can do this by setting the [`credentials`](https://github.com/expressjs/cors#configuration-options) option to `true`:
+
+```js
+//  The example below is using the apollo-server-express package.
+
+// ... set up the server
+await server.start();
+
+const corsOptions = { origin: yourOrigin, credentials: true };
+
+server.applyMiddleware({
+  app,
+  cors: corsOptions,
+  path: "/graphql",
+});
+```
+
+> You *must* specify an origin to enable credentialed requests (i.e., your origin can't use the `*` wildcard value).
```

Comment:
One thing to be careful of with the express `cors` middleware specifically: specifying `{origin: true, credentials: true}` is subtly different from `{origin: '*', credentials: true}` — `origin: true` means "no matter what origin the browser sends, echo it back in the ACAO header" rather than "return ACAO: *" which means that credentials *will* be enabled. So this is rarely safe and should generally be avoided, at least in production.

Reply:
Great call out—I've added a couple of sentences addressing this! :)